tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -294,27 +297,70 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential.FromComponents(
                 mechanism: "SCRAM-SHA-256", source: null, username: uniqueUnicodeUserName, password: unicodePassword);
-            
+
             AssertAuthenticationSucceeds(settings);
         }
 #endif
-        
-        private void AssertAuthenticationSucceeds(MongoClientSettings settings)
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Authentication_succeeds_with_MONGODB_X509_mechanism(
+            [Values(false, true)] bool async)
         {
-             new MongoClient(settings).ListDatabaseNames().ToEnumerable().ToList();
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH", isDefined: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD", isDefined: true);
+            RequireServer.Check().Tls(required: true);
+
+            var pathToClientCertificate = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH");
+            var password = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD");
+            var clientCertificate = new X509Certificate2(pathToClientCertificate, password);
+
+            var userName = GetRfc2253FormattedUsernameFromX509ClientCertificate(clientCertificate);
+            var dropUserResults =
+                DropDatabaseUser(DriverTestConfiguration.Client, userName, databases: new[] { "$external" });
+            var createUserResults = CreateX509DatabaseUser(DriverTestConfiguration.Client, userName);
```

Comment:
Replace previous two lines with:

    _ = DropDatabaseUser(DriverTestConfiguration.Client, "$external", userName);
    _ = CreateX509DatabaseUser(DriverTestConfiguration.Client, userName);

Consider making those private helper methods void since you aren't using the return values.

Reply:
Took the discard version! I feel having the private helper methods return a value could be useful in for future tests or for debugging so I'm inclined to leave it unless people feel very strongly about this.

Reply:
As for me, I think it's not a good sign to have code which is not used right now (I think some exceptions could be but not in this case). Let's discuss it on retro.